lib/tropic/tropic_paragraph.flow

```diff
@@ -296,31 +324,64 @@ reflowTParaWordsRec(
 					TParaWord(word with ghostView = newView);
 				};
 				width = getWordWidth(word);
-				nextWordWidth = if (rest != []) getWordWidth(rest[0]) else 0.0;
+				nextWordWidth = if (rest.words != []) getWordWidth(rest.words[0]) else 0.0;
 
+				addChange = \paraword -> {
+					WordsAndIndexes(
+						arrayPush(wordsChanges.words, paraword),
+						arrayPush(wordsChanges.indexes, indexes[0])
```

Comment:
knowing the initial length of words and length of current words and current length of words we can calculate the index of chenge without passing addition array of indices.

Reply:
> knowing the initial length of words and length of current words and current length of words we can calculate the index of chenge without passing addition array of indices.

I tried this approach, unfortunately it's not accurate because we not always add current words, e.g. https://github.com/area9innovation/flow9/pull/545/files#diff-9b8fade5cbf279ff9a19f9bbc55111a7R318
https://github.com/area9innovation/flow9/pull/545/files#diff-9b8fade5cbf279ff9a19f9bbc55111a7R384

Reply:
@artxp ,investigate it deeper, not passing extra array is good reason for checking